src/backend/cpu/kernel/sift_nonfree.hpp

```diff
@@ -969,6 +969,7 @@ unsigned sift_impl(Array<float>& x, Array<float>& y, Array<float>& score,
                    const bool compute_GLOH)
 {
     in.eval();
+    getQueue().sync();
```

Comment:
The rest of the algorithm is synchronous. This function is not being called in a queue.